### PR TITLE
Implement `fda log` command.

### PR DESCRIPTION
**Pull request by gz**
To retrieve the logs of a pipeline.

### CODE REVIEW

File: crates/fda/src/cli.rs [hunk 1]
```diff
@@ -272,6 +272,18 @@ pub enum PipelineAction {
         #[arg(value_hint = ValueHint::Other, add = ArgValueCompleter::new(pipeline_names))]
         name: String,
     },
+    /// Retrieve the logs of a pipeline.
+    #[clap(aliases = &["log"])]
+    Logs {
+        /// The name of the pipeline.
+        #[arg(value_hint = ValueHint::Other, add = ArgValueCompleter::new(pipeline_names))]
+        name: String,
+        /// Watch the log endpoint and emit new pipeline log messages as they are written.
```
Comment (mihaibudiu) [inline crates/fda/src/cli.rs:281]: this information should be somewhere in the documentation too.
is the rustdoc enough?

**Review by mihaibudiu — APPROVED** [review]

File: crates/fda/src/cli.rs (hunk 1 above)
Comment (gz) [reply crates/fda/src/cli.rs:282]: `fda log --help` will display this

Comment (mihaibudiu) [reply crates/fda/src/cli.rs:282]: can you a README file which is automatically generated from fda --help?

Comment (gz) [reply crates/fda/src/cli.rs:282]: we have this: https://docs.feldera.com/reference/cli I can add the logs command as an example

File: crates/fda/src/main.rs [hunk 2]
```diff
@@ -584,6 +588,50 @@ async fn pipeline(action: PipelineAction, client: Client) {
                     .expect("Failed to serialize pipeline stats")
             );
         }
+        PipelineAction::Logs { name, watch } => {
```
Comment (ryzhyk) [inline crates/fda/src/main.rs:591]: shouldn't this handle the situation when the pipeline isn't running, and hence there are no logs.  And in the watch mode it should exit without an error when the pipeline terminates?

File: crates/fda/src/cli.rs [hunk 3]
```diff
@@ -272,6 +273,18 @@ pub enum PipelineAction {
...
+        ///
+        /// Setting this to true will prevent the command from exiting.
```
Comment (ryzhyk) [inline crates/fda/src/cli.rs:284]:
```suggestion
        /// When `true`, the command will listen to pipeline logs until the pipeline terminates or the command is interrupted by the user. When `false`, the command outputs pipeline logs accumulated so far and exits.
```

**Review by ryzhyk — APPROVED** [review]

File: crates/fda/src/main.rs (hunk 2 above)
Comment (gz) [reply crates/fda/src/main.rs:591]: > shouldn't this handle the situation when the pipeline isn't running

```
❯ cargo run -- log --watch test
     Running `/home/gz/workspace/feldera/target/debug/fda log --watch asof`
LOG STREAM UNAVAILABLE: the pipeline has likely not yet started
❯
```

>  And in the watch mode it should exit without an error when the pipeline terminates

```
2024-09-29 01:14:51 INFO [pipeline-019229e6-8e28-7ef1-880e-cfab04c94d6c] Started HTTP server on port 40119
2024-09-29 01:14:51 INFO [pipeline-019229e6-8e28-7ef1-880e-cfab04c94d6c] Pipeline initialization complete
LOG STREAM END: pipeline is being shutdown

❯ echo $?
0
```

did you try it in some way where it didnt?

**Pull request merged**